src/transformers/models/llava/image_processing_llava.py

```diff
@@ -0,0 +1,438 @@
+            "image_mean",
+            "image_std",
+            "do_convert_rgb",
+            "return_tensors",
+            "data_format",
+            "input_data_format",
+        ]
+
+        # for backwards compatibility of KOSMOS-2
+        if "use_square_size" in kwargs and kwargs["use_square_size"]:
+            self.size = {"height": size["shortest_edge"], "width": size["shortest_edge"]}
+            # Let's remove `use_square_size` (as it is removed from #27690), so the future Kosmos-2 image processors
+            # won't have this attr. being saved. (otherwise, it will enter this if branch while there is no more
+            # `shortest_edge` key.
+            delattr(self, "use_square_size")
+
+    def pad_to_square_original(
+        self, image: Image.Image, background_color: Union[int, Tuple[int, int, int]] = 0
+    ) -> Image.Image:
+        """
+        Pads an image to make it square.
+        """
+        print("Image size:", image.size)
+        width, height = image.size
+        if width == height:
+            return image
+        elif width > height:
+            result = Image.new(image.mode, (width, width), background_color)
+            result.paste(image, (0, (width - height) // 2))
+            return result
+        else:
+            result = Image.new(image.mode, (height, height), background_color)
+            result.paste(image, ((height - width) // 2, 0))
+            return result
+
+    def pad_to_square(
+        self,
+        image: np.array,
+        background_color: Union[int, Tuple[int, int, int]] = 0,
+        input_data_format: Optional[Union[str, ChannelDimension]] = None,
+    ) -> np.array:
+        """
+        Pads an image to a square.
+
+        Args:
+            image (`np.ndarray`):
+                The image to pad.
+            background_color (`int` or `Tuple[int, int, int]`, *optional*, defaults to 0):
+                The color to use for the padding. Can be an integer or a tuple of integers representing the RGB values.
+            input_data_format (`str` or `ChannelDimension`, *optional*):
+                The channel dimension format for the input image. Can be one of:
+                    - `"channels_first"` or `ChannelDimension.FIRST`: image in (num_channels, height, width) format.
+                    - `"channels_last"` or `ChannelDimension.LAST`: image in (height, width, num_channels) format.
+                If unset, will use the inferred format of the input image.
+
+        Returns:
+            `np.ndarray`: The padded image.
+        """
+        height, width = get_image_size(image, input_data_format)
+
+        if height == width:
+            return image
+
+        max_dim = max(height, width)
+        pad_height = max_dim - height
+        pad_width = max_dim - width
+
+        padding = (
+            (pad_height // 2, pad_height - pad_height // 2),
+            (pad_width // 2, pad_width - pad_width // 2),
+        )
+
+        return pad(
```

Comment:
cc @amyeroberts I'd like to replicate [this function](https://github.com/haotian-liu/LLaVA/blob/c121f0432da27facab705978f83c4ada465e46fd/llava/mm_utils.py#L152) using the `pad` function of image_transforms.py. However, it seems padding the channel dimension with custom values isn't supported (they pad using background_color = (122, 116, 104) as seen [here](https://github.com/haotian-liu/LLaVA/blob/c121f0432da27facab705978f83c4ada465e46fd/llava/mm_utils.py#L171)).

Is this something we could add support for?

Reply:
I'd just write a custom `pad` function in this module. If it turns out being a functionality which is nedded for a lot of other models, then we can think about adding into the more general image_transforms module